lib/presentation/widgets/search_list.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:marvel/presentation/features/search/bloc/search_bloc.dart';
+import 'package:marvel/presentation/features/search/bloc/search_event.dart';
+import 'package:marvel/presentation/navigation/app_routes.dart';
+import 'package:marvel/presentation/widgets/bottom_loader.dart';
+import 'package:marvel/presentation/widgets/character_card_widget.dart';
+import 'package:marvel/presentation/widgets/error_page.dart';
+import 'package:marvel/presentation/widgets/search_bottom_error.dart';
+
+class SearchList extends StatefulWidget {
+  final String nameStartsWith;
+
+  const SearchList({Key? key, required this.nameStartsWith}) : super(key: key);
+
+  @override
+  State<SearchList> createState() => _SearchListState();
+}
+
+class _SearchListState extends State<SearchList> {
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_onScroll);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final loading = BlocProvider.of<SearchBloc>(context).state.loading;
+    final characters = BlocProvider.of<SearchBloc>(context).state.charactersViewData;
+    final error = BlocProvider.of<SearchBloc>(context).state.error;
+    final noResult = BlocProvider.of<SearchBloc>(context).state.noResult;
+    final emptySearchField = BlocProvider.of<SearchBloc>(context).state.emptySearchField;
+    final hasReachedMax = BlocProvider.of<SearchBloc>(context).state.hasReachedMax;
+    final afterScroll = BlocProvider.of<SearchBloc>(context).state.afterScroll;
```

Comment:
You must use only State to transfer any data from bloc to your view via Stream, then pass just values to the widgets and do not use Bloc in child widgets. So in you Page you receiving the SearchState, there you take some required fields and pass them through the constructor to the SearchList.

Reply:
done